cmd/zen-go/internal/instrumentor.go

```
@@ -21,27 +21,22 @@ type Instrumentor struct {
 	WrapRules []WrapRule
 }
 
+// NewInstrumentor creates a new Instrumentor, loading rules from YAML files
 func NewInstrumentor() *Instrumentor {
-	return &Instrumentor{
-		WrapRules: []WrapRule{
-			{
-				ID:        "gin.Default",
-				MatchCall: "github.com/gin-gonic/gin.Default",
-				Imports: map[string]string{
-					"zengin": "github.com/AikidoSec/firewall-go/instrumentation/sources/gin-gonic/gin",
-				},
-				WrapTmpl: `func() *gin.Engine { e := {{.}}; e.Use(zengin.GetMiddleware()); return e }()`,
-			},
-			{
-				ID:        "gin.New",
-				MatchCall: "github.com/gin-gonic/gin.New",
-				Imports: map[string]string{
-					"zengin": "github.com/AikidoSec/firewall-go/instrumentation/sources/gin-gonic/gin",
-				},
-				WrapTmpl: `func() *gin.Engine { e := {{.}}; e.Use(zengin.GetMiddleware()); return e }()`,
-			},
-		},
+	// Try to load rules from the instrumentation directory
+	if instDir := findInstrumentationDir(); instDir != "" {
+		if rules, err := loadRulesFromDir(instDir); err == nil {
+			return &Instrumentor{WrapRules: rules}
+		}
```

error being swallowed?